### PR TITLE
Misc › Additional config options

### PR DESCRIPTION
**Pull request by jpirkey**
Adding additional config options:

1) buttonLabel for overriding the default button label in the toolbar
2) uploadCallback for when a file has been successfully uploaded

### CODE REVIEW

File: uploadcare.js [hunk 1]
```diff
@@ -49,6 +49,10 @@ if (!RedactorPlugins) var RedactorPlugins = {};
                         } else {
                             $this.insert.html('<a href="' + this.cdnUrl + '">' + info.name + '"</a>', false);
                         }
+
+                        if ($.isFunction($opts.uploadCallback)) {
```
Comment (dmitry-mukhin) [inline uploadcare.js]: `uploadCompleteCallback` seems to be more precise name for this.
what do you think?

**Comment by jpirkey** [comment]
Completely fine with that.  I think though I want a more thorough callback that handles the inserting of the preview, since I need to copy it to our storage first, then provide the correct CDN.  Another reason for this  is that you guys return `http` links which won't work on our domain since we force SSL for everything.  So I am going to rework this a bit and get back to you.

**Comment by dmitry-mukhin** [comment]
Well, all this happens when the download is complete though :)
As for scheme, there is [this](https://uploadcare.com/documentation/widget/#option-cdn-base) option.

**Comment by jpirkey** [comment]
Exactly; which is why I am thinking that it should be do your existing logic if `uploadCompleteCallback` isn't defined or it is up to the `uploadCompleteCallback` to do the inserting or whatever else it needs to do (like copying it to our storage).  Sometimes easier to explain in code:

```
done: function(data) {
    var $this = this;
    var files = $opts.multiple ? data.files() : [data];
    $.when.apply(null, files).done(function() {
        $.each(arguments, function() {
            if ($.isFunction($opts.uploadCompleteCallback)) {
                $opts.uploadCompleteCallback.call($this, this);
            } else {
                var imageUrl = this.cdnUrl;
                if (this.isImage && ! this.cdnUrlModifiers) {
                    imageUrl += '-/preview/';
                }
                if (this.isImage) {
                    $this.insert.html('<img src="' + imageUrl + '" alt="' + this.name + '" />', false);
                } else {
                    $this.insert.html('<a href="' + this.cdnUrl + '">' + info.name + '"</a>', false);
                }
            }
        });
    });
},
```

**Comment by dmitry-mukhin** [comment]
I see, this indeed seems to be a better way to do things.

**Comment by jpirkey** [comment]
Submitted update.

**Comment by dmitry-mukhin** [comment]
thanks!

**Pull request merged**